### PR TITLE
Feature/react native

### PR DESCRIPTION
**Pull request by tragessere**
Used Q promises to support returned promises from storage service as react native does not allow synchronous access to storage.

### CODE REVIEW

**Comment by builtbywill** [comment]
@tragessere I just had the thought and wanted to clarify, I had only used "q" because it was included in the caliper-js package.json when I started and I didn't want to make too many changes. Since we are already moving forward with our branch a bit, we might want to consider using a different promise lib like "bluebird". https://bytearcher.com/articles/q-or-bluebird/

@jkdufair what are your thoughts?

File: lib/caliper-service.js [hunk 1]
```diff
@@ -171,28 +177,43 @@ function CaliperService(options) {
...
 	 */
 	function loadSavedData() {
-		authToken = _options.storageService.getItem(authTokenKey) || null;
-		person = _options.storageService.getItem(personKey) || null;
-		session = _options.storageService.getItem(sessionKey) || null;
-		sessionExtensions = _options.storageService.getItem(sessionExtensionsKey) || null;
-		lastKeepAliveDate = _options.storageService.getItem(lastKeepAliveDateKey) || null;
-		lastActivityDate = _options.storageService.getItem(lastActivityDateKey)
+		return Q.all([
+			_options.storageService.getItem(authTokenKey),
```
Comment (builtbywill) [inline lib/caliper-service.js]: are you assuming the storage service methods return promises? what happens here if they do not?

File: lib/caliper-service.js [hunk 2]
```diff
@@ -468,7 +489,7 @@ function CaliperService(options) {
 	function addActivityListeners() {
 		/*eslint no-undef: "error"*/
 		/*eslint-env browser*/
-		if (typeof window === 'undefined') {
+		if (typeof window === 'undefined' || !(typeof window.addEventListener === 'function')) {
```
Comment (builtbywill) [inline lib/caliper-service.js]: you can probably just use `!==` and ditch the extra parentheses

File: lib/caliper-service.js (hunk 1 above)
Comment (tragessere) [inline lib/caliper-service.js]: Anything that isn't a promise just gets passed as part of the values array as if it were the resolved value of a promise. Basically it behaves the same whether all, none, or any combination of values are actually promises.

**Pull request merged**